backend/lib/endpoints/posts.js

```diff
@@ -207,19 +216,49 @@ async function routes(app) {
           },
         },
       ];
+      /* eslint-enable sort-keys */
+
+      const aggregationPipelineResults = [
+        ...sortAndFilterSteps,
+        ...paginationSteps,
+        ...lookupSteps,
+        ...projectionSteps,
+      ];
+
+      // Get the total results without pagination steps but with filtering aplyed - totalResults
+      /* eslint-disable sort-keys */
+      const totalResultsAggregationPipeline = await Post.aggregate([
+        ...sortAndFilterSteps,
+        ...lookupSteps,
+        { $group: { _id: null, count: { $sum: 1 } } },
+      ]);
+      /* eslint-enable sort-keys */
+
+      const totalResults = totalResultsAggregationPipeline[0].count;
```

Comment:
```suggestion
```


This will throw an error if there are no results - []

```json
{
    "statusCode": 500,
    "error": "Internal Server Error",
    "message": "Cannot read property 'count' of undefined"
}
```

My suggestion would be to skip this line & return something like...

```javascript
return {
  meta: { total: totalResultsAggregationPipeline.length ? totalResultsAggregationPipeline[0].count : 0 },
  data: response
}
```

Reply:
Thanks again for your review Josh. Commited the changes.